final_task/rss_reader/feed.py

```diff
@@ -0,0 +1,89 @@
+import logging
+from string_operations import make_string_readable
+from pprint import pprint
+import article
+import shelve
+import check_func as check
+
+class Feed:
+    """Feed class, contain feed info and list of articles """
+    def __init__(self, parsed, args):
+        """create feed with fixed number of articles """
+        logging.info('Started creting feed')
+
+        articles_list = []
+        cashed_news_number = 0
+        if args.date:
+            logging.info('Started extracting data from cash')
+            check.check_data_base('cashed_feeds')
+            self.link = args.source
+            self.feed_name = 'Feeds from {}'.format(args.source)
+            with shelve.open('cashed_feeds') as database:
+                check.is_date_in_database(args.date, 'cashed_feeds')
```

Comment:
You are trying to open a database in your `is_date_in_database` function while it is actually already opened here. This may cause problems.

Reply:
Fixed, thanks